### PR TITLE
[Feature-WIP](multi-catalog) Add Hive sink on BE side.

### PR DESCRIPTION
**Pull request by kaka11chen**
## Proposed changes

Issue Number: #31442

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/sink/writer/vhive_table_writer.cpp [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+
+    return std::make_shared<VHivePartitionWriter>(
+            _t_sink, std::move(partition_name), update_mode, _vec_output_expr_ctxs,
+            hive_table_sink.columns, std::move(write_info),
+            fmt::format("{}{}", _compute_file_name(),
+                        _get_file_extension(file_format_type, write_compress_type)),
+            file_format_type, write_compress_type, hive_table_sink.hadoop_config);
+}
+
+std::vector<std::string> VHiveTableWriter::_create_partition_values(vectorized::Block& block,
```
Comment (morningman) [inline be/src/vec/sink/writer/vhive_table_writer.cpp:242]: give some example this this convertor

**Comment by kaka11chen** [comment]
run buildall

**Comment by kaka11chen** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by zy-kkk — APPROVED** [review]

**Pull request merged**